lib/engine.py

```diff
@@ -135,32 +139,34 @@ def source_gen():
                 except Exception as e:
                     self.warning('Exception in line %s: %s' % (self.table.record_id, e))
                     continue
-                try:
-                    insert_stmt = self.insert_statement(cleanvalues)
-                except:
-                    if self.debug:
-                        print(types)
-                    if self.debug:
-                        print(linevalues)
-                    if self.debug:
-                        print(cleanvalues)
-                    raise
-                try:
-                    update_frequency = int(self.update_frequency)
-                except:
-                    update_frequency = 100
-
-                if (self.table.record_id % update_frequency == 0 or
-                        self.table.record_id == 1 or
-                        self.table.record_id == total):
-                    prompt = "Inserting rows to " + self.table_name() + ": "
-                    prompt += str(self.table.record_id) + " / " + str(total)
-                    sys.stdout.write(prompt + "\b" * len(prompt))
-                try:
-                    self.execute(insert_stmt, commit=False)
-                except:
-                    print(insert_stmt)
-                    raise
+
+                if count_iter % insert_limit == 0 or count_iter == real_line_length:
+                    multiple_values.append(cleanvalues)
+                    try:
+                        insert_stmt = self.insert_statement(multiple_values)
+                    except:
+                        if self.debug:
+                            print(types)
+                        if self.debug:
+                            print(linevalues)
+                        if self.debug:
+                            print(cleanvalues)
+                        raise
+                    multiple_values = []
+
+                    try:
+                        self.execute(insert_stmt, commit=False)
+                        current += insert_limit
+                        if current > real_line_length:
+                            print(str(real_line_length) + " rows inserted into " + self.table_name() + ": ")
+                        else:
+                            print(str(current) + " rows inserted into " + self.table_name() + ": ")
+                    except:
+                        print(insert_stmt)
+                        raise
+                else:
+                    multiple_values.append(cleanvalues)
+                count_iter += 1
```

Comment:
This issues dates back before either this PR or @henrykironde, but `gen_source` is definitely too long and it makes reviewing changes to it hard work. Definitely something to target for refactoring sometime soon.